### PR TITLE
Allow users to download their own specified sensor version

### PR DESCRIPTION
**Pull request by isimluk**

### CODE REVIEW

File: examples/falcon_sensor_download/main.go [hunk 1]
```diff
@@ -64,11 +66,21 @@ Falcon Client Secret`)
...
 		}
-		sensor := querySuitableSensor(client, *osName, *osVersion)
+		if *sensorVersion == "" {
+			validSensorVersions := getValidSensorVersions(client, *osName, *osVersion)
+			if len(validSensorVersions) == 0 {
+				fmt.Fprintf(os.Stderr, "No sensors available for specified os/version tuple: %s/%s\n", *osName, *osVersion)
+				os.Exit(1)
+			}
+			fmt.Printf("Missing --sensor-version=latest command-line option. Available sensor versions are: [%s]\n", strings.Join(validSensorVersions, ", "))
+			*sensorVersion = promptUser("Selected Sensor Version")
```
Comment (shawndwells) [inline examples/falcon_sensor_download/main.go:80]:
```suggestion
			fmt.Printf("Missing --sensor-version command-line option. Available sensor versions are: [%s]\n", strings.Join(validSensorVersions, ", "))
			*sensorVersion = promptUser("Selected Sensor Version")
```

**Review by shawndwells — CHANGES_REQUESTED** [review]

**Comment by isimluk** [comment]
The default is the latest.

**Comment by isimluk** [comment]
```
$ go run ./examples/falcon_sensor_download --help                                      
Usage of /var/folders/fp/2wb2sw_n0q97hd4ymwvnrbs00000gn/T/go-build946107107/b001/exe/falcon_sensor_download:
  -all
    	Download all sensors
  -client-id string
    	Client ID for accessing CrowdStrike Falcon Platform (default taken from FALCON_CLIENT_ID env) (default "9e2c9d74782540f79fad17c516468315")
  -client-secret string
    	Client Secret for accessing CrowdStrike Falcon Platform (default taken from FALCON_CLIENT_SECRET) (default "8VuSnk2pAs01QwdU537cT4KW9eOyXEbIDRHxPZ6M")
  -cloud string
    	Falcon cloud abbreviation (us-1, us-2, eu-1, us-gov-1)
  -os-name string
    	Name of the operating system
  -os-version string
    	Versin of the operating system
  -sensor-version string
    	Version of the Falcon Sensor. Use: 'latest' to get the latest or '' to get prompted interactively (default "latest")
```

**Comment by isimluk** [comment]
```
$ go run ./examples/falcon_sensor_download --help                                      
Usage of /var/folders/fp/2wb2sw_n0q97hd4ymwvnrbs00000gn/T/go-build946107107/b001/exe/falcon_sensor_download:
  -all
    	Download all sensors
  -client-id string
    	Client ID for accessing CrowdStrike Falcon Platform (default taken from FALCON_CLIENT_ID env) (default "")
  -client-secret string
    	Client Secret for accessing CrowdStrike Falcon Platform (default taken from FALCON_CLIENT_SECRET) (default "")
  -cloud string
    	Falcon cloud abbreviation (us-1, us-2, eu-1, us-gov-1)
  -os-name string
    	Name of the operating system
  -os-version string
    	Versin of the operating system
  -sensor-version string
    	Version of the Falcon Sensor. Use: 'latest' to get the latest or '' to get prompted interactively (default "latest")
```

File: examples/falcon_sensor_download/main.go (hunk 1 above)
Comment (isimluk) [reply examples/falcon_sensor_download/main.go:80]: Unfortunately this won't work.

There are multiple compromises that needs to be made here to allow for the least possible amount of surprises. As a result the message in questions (that is rarely to be seen) contains some surprises.

**Comment by isimluk** [comment]
I have reached a belief that it may be worth to kindly notify reviewers of this PR to revisit and see if their previous conclusions however well argued could have been mistaken.

**Review by shawndwells — APPROVED** [review]

**Comment by shawndwells** [comment]
Indeed! Thanks for the nudge.

**Comment by redhatrises** [comment]
hahahahaha

**Pull request merged**